Transactions.md

```diff
@@ -1,182 +1,291 @@
-### <a name="ClosingKnownValues">Known Values</a>
-  * Local Address: `BitcoinAddress`
-  * Remote's ToLocalOutput: `ScriptPubKey`
-  * Fee Rate: `FeeUnit`
-
-Where
-  - `Local Address` is any unused local address
-  - `Remote's ToLocalOutput` is of the form [specified above](#CETOutputs)
-### <a name="ClosingInputs">Inputs</a>
-  * Input Spending(P2WSH(Remote's ToLocalOutput))
-### <a name="ClosingOutputs">Outputs</a>
-  * One output corresponding to `LocalAddress` with value `P2WSH(Remote's ToLocalOutput).value - fee`
+This details the exact format of all on-chain transactions, which both sides need to agree on to ensure signatures are valid.
 
+# Table of Contents
+
+* [Funding Transaction](#funding-transaction)
+  * [Funding Transaction Input and Output Ordering](#funding-transaction-input-and-output-ordering)
+  * [Funding Inputs](#funding-inputs)
+  * [FundingOutput](#funding-output)
+  * [Change Outputs](#change-outputs)
+* [Contract Execution Transaction](#contract-execution-transaction)
+  * [CET Outputs](#cet-outputs)
+    * [Offerer Output](#offerer-output)
+    * [Accepter Output](#accepter-output)
+* [Refund Transaction](#refund-transaction)
+* [Fees](#fees)
+  * [Expected Weight of the Funding Transaction](#expected-weight-of-the-funding-transaction)
+  * [Expected Weight of the Contract Execution or Refund Transaction](#expected-weight-of-the-contract-execution-or-refund-transaction)
+* [Test Vectors](#test-vectors)
+* [References](#references)
+* [Authors](#authors)
+
+# Funding Transaction
+
+* version: 2
+* locktime: 0
+
+The funding inputs and change output script public keys are negotiated in the offer and accept messages.
+
+## Funding Transaction Input and Output Ordering
+
+The Offerer/Initiator's inputs come first and in the order that they were listed in the Offer message, the Accepter's inputs come after this in the order they are listed in the Accept message. The funding output comes first, followed by the Offerer's change output if applicable, and then the Accepter's change output if applicable.
+
+Note that this will likely change in the future.
+
+## Funding Inputs
+
+All funding inputs must be Segwit or nested P2SH(Segwit) in order to protect against malleability attacks.
+
+## Funding Output
+
+* The funding output script is a P2WSH to:
+
+```
+2 <offer_funding_pubkey> <accept_funding_pubkey> 2 OP_CHECKMULTISIG
+```
+
+* Where both `pubkey`s are in compressed format.
+
+## Change Outputs
+
+The funding transaction's change outputs should pay to the address specified in the relevant offer/accept message. A change output's value should equal the total funding amount of that party subtracted by their total collateral, their fees for this transaction, and their fees for the largest possible closing transaction. If this value is below the dust limit of `1000 satoshis`, then that party must include additional funding in order to ensure they have a valid anchor output.
```

Comment:
Ah I see you already wrote the requirement. Should we copy-paste anchor output scripts and requirements from BOLT or referring them, especially with regards to utxo set pollution prevention ? At least you should mention to not set the witnessScript with some third-party spendable path to avoid this one interfering with your funding confirmation.

Reply:
See https://github.com/discreetlogcontracts/dlcspecs/issues/54#issuecomment-675853144

I don't think we need to actually specify anything for anchor outputs

Reply:
@benthecarman You mean we consider DLC parties to always spend their change outputs even when it's a low-economic value of 1000 satoshis ? 

It just a blockchain usage gentleman's rule to avoid utxo set pollution in case of multi-party protocol, we don't have to hold to it.

Reply:
Not that they always spend them but they have the option to for CPFP